### PR TITLE
Remove flaky test

### PR DESCRIPTION
**Pull request by BrennanConroy**
Talked with @Tratcher and we think the test wasn't doing much and is flaky, so we're deleting it.

### CODE REVIEW

**Review by Tratcher — APPROVED** [review]
https://github.com/aspnet/AspNetCore-Internal/issues/3019
https://github.com/aspnet/AspNetCore-Internal/issues/3128

File: src/Servers/HttpSys/test/FunctionalTests/ServerTests.cs [hunk 1]
```diff
@@ -366,31 +366,6 @@ public void Server_SetConnectionLimitArgumentValidation_Success()
...
-                using (var client2 = await SendHungRequestAsync("GET", address))
-                {
-                    using (var client3 = await SendHungRequestAsync("GET", address))
-                    {
-                        // Maxed out, refuses connection and throws
-                        await Assert.ThrowsAsync<HttpRequestException>(() => SendRequestAsync(address));
-                    }
-
-                    // A connection has been closed, try again.
-                    string responseText = await SendRequestAsync(address);
```
Comment (halter73) [inline src/Servers/HttpSys/test/FunctionalTests/ServerTests.cs]: Would this test be reliable if we removed just this last request? Looking at the [last failure](https://dnceng.visualstudio.com/internal/_build/results?buildId=397114&view=ms.vss-test-web.build-test-results-tab&runId=12397114&resultId=119016&paneView=debug), that seems to be where the problem is.

Do we have any other tests that verifies `HttpSysOptions.MaxConnections` actually does anything?

Comment (halter73) [reply src/Servers/HttpSys/test/FunctionalTests/ServerTests.cs]: If we stopped passing the option to HttpSys, would any tests fail other than this one we're removing?

Comment (Tratcher) [reply src/Servers/HttpSys/test/FunctionalTests/ServerTests.cs]: No.

Comment (BrennanConroy) [reply src/Servers/HttpSys/test/FunctionalTests/ServerTests.cs]: > Do we have any other tests that verifies `HttpSysOptions.MaxConnections` actually does anything?

Yes

**Pull request merged**